apps/web/playwright/reschedule.e2e.ts

```diff
@@ -239,4 +240,27 @@ test.describe("Reschedule Tests", async () => {
     expect(newBooking).not.toBeNull();
     expect(newBooking?.status).toBe(BookingStatus.ACCEPTED);
   });
+
+  test("Should be able to book slot that overlaps with original rescheduled booking", async ({
+    page,
+    users,
+    bookings,
+  }) => {
+    const user = await users.create();
+    const eventType = user.eventTypes[0];
+    const startTime = dayjs().add(1, "day").set("hour", 10).set("minute", 0).toDate();
+    const endTime = dayjs().add(1, "day").set("hour", 10).set("minute", 30).toDate();
+
+    const booking = await bookings.create(user.id, user.username, eventType.id, {}, startTime, endTime);
+
+    await page.goto(`/reschedule/${booking.uid}`);
+
+    //book same slot again
+    page.getByRole("button", { name: dayjs(startTime).format("D") }).click();
```

Comment:
Seems like we're getting 5 buttons with the same startTime

<img width="775" alt="image" src="https://github.com/calcom/cal.com/assets/3504472/33c31ae1-71f7-4ec7-8bc5-1dd4e523cc53">

Reply:
fixed 👍